tests/test_views.py

```diff
@@ -0,0 +1,1050 @@
+            }
+        ),
+        (
+            True,
+            {'scope': 1, 'nominator_name': 'John Doe', 'nominator_institution': 'UNT'},
+            {'nominator_email': u'This field is required.'}
+        )
+    ])
+    def test_context_with_post_and_errors(self, client, reg_required, data, expected_errors):
+        entity = 'www.example.com'
+        project = factories.ProjectFactory(registration_required=reg_required)
+        factories.URLFactory(url_project=project, entity=entity)
+        response = client.post(
+            reverse('url_listing', args=[project.project_slug, entity]),
+            data
+        )
+
+        assert response.context['form_errors'] == expected_errors
+        assert response.context['summary_list'] is None
+        assert response.context['scope_form'].is_valid() is False
+        assert sorted(json.loads(response.context['json_data'])) == sorted(
+            [[key, [str(data[key])]] for key in data])
+
+    def test_context_with_get(self, client):
+        entity = u'http://www.example.com'
+        entity_surt = u'http://(com,example,www,)'
+        project = factories.ProjectFactory()
+        factories.ProjectMetadataFactory(project=project)
+        factories.URLFactory(url_project=project, entity=entity)
+        factories.SURTFactory(
+            url_project=project,
+            entity=entity,
+            value=entity_surt
+        )
+        factories.SURTFactory.create_batch(
+            1,
+            url_project=project,
+            entity=u'{}/main/page'.format(entity),
+            value=u'{}/main/page'.format(entity_surt)
+        )
+
+        expected_context = {
+            'scope_form': views.ScopeForm(),
+            'form_errors': None,
+            'summary_list': None,
+            'json_data': None
+        }
+        response = client.get(reverse('url_listing', args=[project.project_slug, entity]))
+
+        for key in expected_context:
+            assert str(response.context[key]) == str(expected_context[key])
+
+
+class TestUrlSurt():
+
+    def test_status_ok(self, rf):
+        project = factories.ProjectFactory()
+        request = rf.get('/')
+        response = views.url_surt(request, project.project_slug, 'a_surt')
+
+        assert response.status_code == 200
+
+    def test_template_used(self, client):
+        project = factories.ProjectFactory()
+        response = client.get(reverse('url_surt', args=[project.project_slug, 'a surt']))
+        assert response.templates[0].name == 'nomination/url_surt.html'
+
+    @pytest.mark.parametrize('surt_root, surt, url_list_len, letter', [
+        ('http://(com,e', 'http://(com,example,www)', 1, 'e'),
+        ('http://(com,example,www)', 'http://(com,example,www)', 1, False),
+        ('http://(com,a', 'http://(com,example,www)', 0, 'a'),
+        ('http://(com,apples,www)', 'http://(com,example,www)', 0, False)
+    ])
+    def test_context(self, client, surt_root, surt, url_list_len, letter):
```

Comment:
It seems to be that we are maybe testing more than we should here. The only thing that should change based on the input is the `browse_domain` context var, correct?